template/ut_template/bin/run.sh

```diff
@@ -19,6 +19,46 @@
 # * limitations under the License.
 # *
 
+# Save the original directory where the script was called from
+original_dir=$(pwd)
+
+# Function to convert a relative path into an absolute path
+# If the path is already absolute, it remains unchanged
+convert_to_absolute_path() {
+    if [[ "$1" = /* ]]; then
+        # If it's an absolute path, just return it
+        echo "$1"
+    else
+        # If it's a relative path, prepend the original directory to it
+        echo "$original_dir/$1"
+    fi
+}
+
+# Loop through arguments to find the profile path (-p) and other arguments
+profile_path=""
+other_args=()
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -p)  # When -p flag is found, resolve the path that follows it
+            shift
+            profile_path=$(convert_to_absolute_path "$1")
+            ;;
+        *)   # Collect other arguments
+            other_args+=("$1")
+            ;;
+    esac
+    shift
+done
+
+# Change to the directory where the script is located
 cd "$(dirname "$0")"
-export LD_LIBRARY_PATH=/usr/lib:/lib:/home/root
-./hal_test $@
+
+# Set necessary library paths
+export LD_LIBRARY_PATH=/usr/lib:/lib:/home/root:./
+
+# Run the actual program with the resolved absolute profile path and any other arguments
+./hal_test "${other_args[@]}" -p "$profile_path"
```

Comment:
do you think it be safer to swap these, so that -p always first, and if they submit crappy args on the end we can ignore them from the hal_test? Rather than possibly breaking -p because of the args before it?

Reply:
sure, I agree